src/components/BackButton.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import { ArrowBackIcon } from "@chakra-ui/icons";
+import { HStack, Button, Link } from "@chakra-ui/react";
+import { useRouter } from "next/router";
+
+type BackButtonProps = {
+    stypeProps: Record<string, unknown>;
+    onClick?: React.MouseEventHandler;
+};
+
+export const BackButton: React.FC<BackButtonProps> = (props) => {
+    const router = useRouter();
+    const onClick = props.onClick ? props.onClick : () => router.back();
+
+    return (
+        <HStack spacing={6}>
+            <Link onClick={onClick}>
```

Comment:
btw does this need to be wrapped in a `Link` tag or can we just move the `onClick` prop to the `Button` instead?

Reply:
same with `HStack`